pkgs/racket-doc/scribblings/reference/contracts.scrbl

```diff
@@ -1784,6 +1784,16 @@ The @racket[define-struct/contract] form only allows a subset of the
   
 }
 
+@defform[#:literals (current-input-port)
+         (read/contract contract-expr [in (current-input-port)])
+         #:contracts ([in input-port?])]{
+  Calls @racket[read] to retrieve a single @tech{datum} from @emph{in}, then applies
+  @racket[contract-expr] to the result.
```

Comment:
What's the easiest way to add examples? Create a namespace with `racket/port` and call `with-input-from-string`?